### PR TITLE
Add label parameter to Resource object

### PR DESCRIPTION
**Pull request by RoniKishner**
##### Short description:
Add label parameter to Resource class

##### More details:
- Added label parameter to Resource object so it will be added to the resource manifest under metadata->labels
- Changed namespace.py so it will use the new label parameter in Resource object

##### What this PR does / why we need it:
Label field is relevant to all resources, this will enable easier use of creating a resource with labels.
Also made changes to Namespace so they will use the label parameter 

##### Which issue(s) this PR fixes:
Enable use of label when creating a new DataSource

### CODE REVIEW

File: ocp_resources/data_source.py [hunk 1]
```diff
@@ -17,6 +17,7 @@ def __init__(
         teardown=True,
         yaml_file=None,
         delete_timeout=TIMEOUT_4MINUTES,
+        label=None,
```
Comment (rnetser) [inline ocp_resources/data_source.py:20]: either you add it to all resources or do not add it as we already have `kwargs` which are passed.

**Review by rnetser — CHANGES_REQUESTED** [review]

**Review by rnetser — APPROVED** [review]

**Review by rnetser — CHANGES_REQUESTED** [review]
Please update the title and description to match the updated code

**Review by rnetser — APPROVED** [review]

**Review by myakove — APPROVED** [review]

**Pull request merged**